src/Connection.cpp

```diff
@@ -289,15 +289,18 @@ void Connection::emitInterfacesRemovedSignal( const std::string& objectPath
     SDBUS_THROW_ERROR_IF(r < 0, "Failed to emit InterfacesRemoved signal", -r);
 }
 
-SlotPtr Connection::registerSignalHandler( const std::string& objectPath
+SlotPtr Connection::registerSignalHandler( const std::string& sender
+                                         , const std::string& objectPath
                                          , const std::string& interfaceName
                                          , const std::string& signalName
                                          , sd_bus_message_handler_t callback
                                          , void* userData )
 {
     sd_bus_slot *slot{};
 
-    auto filter = composeSignalMatchFilter(objectPath, interfaceName, signalName);
+    // alternatively to our own composeSignalMatchFilter() implementation, we could also use sd_bus_match_signal() from
+    // https://www.freedesktop.org/software/systemd/man/sd_bus_add_match.html#
```

Comment:
Don't we want do it now?

Reply:
It was just the smaller change but it's probably better to use the convenience function if it is already provided by the underlying sdbus lib. Especially if it would have avoided this bug. 

We state that the minimum libsystemd version is v236. Let me check if this function is there in v236. If yes, I will update the code.

Reply:
v237 introduced sd_bus_match_signal(): https://github.com/systemd/systemd/commit/b423e4fb73866e529869b348efb7169ee91f00c9

I don't think this justifies to increase the libsystemd dependency from v236 to v237. The comment was updated but under this circumstances I'd prefer to keep as is.

Reply:
Okay, agreed then.